src/main/java/org/mockito/internal/junit/DefaultStubbingLookupListener.java

```diff
@@ -57,8 +57,11 @@ public void onStubbingLookup(StubbingLookupEvent event) {
         List<Invocation> matchingStubbings = new LinkedList<Invocation>();
         for (Stubbing s : stubbings) {
             if (UnusedStubbingReporting.shouldBeReported(s)
-                && s.getInvocation().getMethod().getName().equals(invocation.getMethod().getName())) {
-                matchingStubbings.add(s.getInvocation());
+                && s.getInvocation().getMethod().getName().equals(invocation.getMethod().getName())
+                //If stubbing and invocation are in the same source file we assume they are in the test code,
+                // and we don't flag it as mismatch:
+                && !s.getInvocation().getLocation().getSourceFile().equals(invocation.getLocation().getSourceFile())) {
```

Comment:
This would not work if the setup is performed in a base class or helper class. Can we figure out a solution for these situations as well?

Reply:
This PR is improving the current behavior a lot already. We can always improve it further without blocking this PR. Feel free to suggest a solution!

Reply:
@TimvdLippe, thoughts?

Reply:
I have not been able to come up with a proper solution, other than using a different stubbing mechanism. I fear that, no matter how hard we try, we will keep on receiving issues on this feature, where users will run into all the edge-cases.

That said, this does improve the current situation, so we can push forward. I just think we have to solve the fundamental problem first to be able to prevent these kind of issues popping up.

Reply:
Thank you for review!